src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.movie_new.service;
+
+import com.exadel.movie_new.model.Movie;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.stereotype.Service;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+
+    @Override
+    public Movie addMovie(Movie movie) {
+        List<Movie> movieData = readJsonData();
+        Movie moviePresent = null;
+
+        if (movieData != null) {

```

Comment:
movieData could be just empty (no films yet). Pay attention to this case

Reply:
This one still relevant. You must understand that in case you have json file but this file empty - your movieData will not be NULL but it will be emptyList. So instead of using check for null you must check for null AND not empty:  if (movieData != null && !movieData.isEmpty())